pkg/cli/cli.go

```diff
@@ -187,6 +189,12 @@ func (c *cli) getInfoFromFlags() (string, []string, error) {
 		plugins[i] = strings.TrimSpace(key)
 	}
 
+	// For backwards compatibility reasons, defining the project version flag and not defining
+	// the plugins flag should be interpreted as if the plugins flag was set to go/v2.
+	if projectVersion == "2" && len(plugins) == 0 {
+		plugins = append(plugins, plugin.KeyFor(pluginv2.Plugin{}))
+	}
+
```

Comment:
we need to update the first comment as well to clarifies that. 
I am ok with this change here since we are ensuring the `backwards compatibility` and not change the behaviour for `kb/sdk init project-version=2`. 


However, for we are able to move forward and get it merged we also need to check it with @estroz.
@estroz wdyt? Is it makes sense for you or do you see any edge scenario that would be affected as well? 

This change is required to address 1.5, however, I'd like to confirm with you that it can be changed to address the needs without effect any requirement.

Reply:
PR description updated. It previously had a behavior that differ from the v2.3.2 release but it no longer does.

Just a note about Camila's comment. This PR applies the breaking changes that will be needed for #1941 and #1942. By doing them right now we ensure that #2019 (which solves #1941 implementing the proposal #1942) will be non-breaking, and therefore wil no longer block v3.0.0 release. (#1991 would still be breaking but at least it is one step less blocking the release.)

Reply:
After re-think here, imo if a plugin supports n project versions than we should still able to use it without inform the --project-version. See: https://github.com/kubernetes-sigs/kubebuilder/pull/1942/files#r584166217

Reply:
You can still define a default project version, that has not changed. The only difference is that you no longer have multiple sets of default plugins.